### PR TITLE
add additional arm64 checker in cmake files

### LINKED ISSUE #48873: aarch64 CMake checks are incorrect on Apple Silicon
Pytorch's CMakeLists.txt files have several checks for `CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64"` that don't take into account the fact that `CMAKE_SYSTEM_PROCESSOR` can be `arm64` on Apple Silicon (or `arm64e` for Pointer Authentication).

As of 6ab84ca0f3458924e79f0279521a8b5720d3a9e3, here are the CMake checks I found that seem to include `aarch64` but not `arm64`:

```
CMakeLists.txt:681
CMakeLists.txt:698
cmake/External/nnpack.cmake:30
cmake/Modules_CUDA_fix/upstream/FindCUDA.cmake:722
third_party/NNPACK/CMakeLists.txt:54
third_party/NNPACK/CMakeLists.txt:88
third_party/QNNPACK/CMakeLists.txt:40
third_party/cpuinfo/CMakeLists.txt:513
third_party/cpuinfo/CMakeLists.txt:762
third_party/cpuinfo/CMakeLists.txt:810
third_party/fbgemm/third_party/cpuinfo/CMakeLists.txt:163
third_party/fbgemm/third_party/cpuinfo/CMakeLists.txt:501
third_party/fbgemm/third_party/cpuinfo/CMakeLists.txt:70
third_party/fbgemm/third_party/cpuinfo/CMakeLists.txt:750
third_party/fbgemm/third_party/cpuinfo/CMakeLists.txt:798
third_party/sleef/Configure.cmake:463
third_party/sleef/src/libm-tester/CMakeLists.txt:35
third_party/sleef/travis/toolchain-aarch64.cmake:3
```

I think this at minimum breaks the NNPACK build checks.

cc @malfet @seemethere @walterddr

### PR DESCRIPTION
**Pull request by walterddr**
tentatively fixes #48873

### CODE REVIEW

File: cmake/Modules_CUDA_fix/upstream/FindCUDA.cmake [hunk 1]
```diff
@@ -719,7 +719,7 @@ if(CMAKE_CROSSCOMPILING)
   elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "arm")
     # Support for arm cross compilation
     set(CUDA_TOOLKIT_TARGET_NAME "armv7-linux-gnueabihf")
-  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
+  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64|arm64)$")
```
Comment (malfet) [inline cmake/Modules_CUDA_fix/upstream/FindCUDA.cmake:722]: I don't think we should worry about CUDA support on ARM Mac
```suggestion
  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -678,7 +678,7 @@ if(USE_ASAN)
     string(APPEND CMAKE_LINKER_FLAGS_DEBUG " -fsanitize=address")
 endif()
 
-if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64|arm64)$")
```
Comment (malfet) [inline CMakeLists.txt:681]: Not needed here, as `clang` for M1 supports VST1 intrinsics
```suggestion
if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -695,7 +695,7 @@ int main() {
   endif()
 endif()
 
-if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64|arm64)$")
```
Comment (malfet) [inline CMakeLists.txt:698]: Same as above
```suggestion
if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
```

**Review by malfet — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (walterddr) [reply CMakeLists.txt:681]: you mean if we use the preferred compiler `clang` in MacOS (on M1), the source compile check is unnecessary? (in this case CMAKE_SYSTEM_PROCESSOR is `arm64` yes?)

File: cmake/Modules_CUDA_fix/upstream/FindCUDA.cmake (hunk 1 above)
Comment (walterddr) [reply cmake/Modules_CUDA_fix/upstream/FindCUDA.cmake:722]: yes. this should be removed.

**Pull request closed without merging**